doc/source/user/api_stability.rst

```diff
@@ -0,0 +1,57 @@
+API Stability
+=============
+
+All public functionality in scikit-bio has a defined stability state.
+These states inform users and developers to what extent they can rely on
+different APIs in the package.
+
+You can find out the stability state of public functionality by looking at its
```

Comment:
Can you add a link to numpydoc here and note that our docstrings follow this format? Then your references to short/extended summaries will be clearer.